pkg/rack/terraform.go

```diff
@@ -196,7 +201,23 @@ func (t Terraform) Metadata() (*Metadata, error) {
 	}
 
 	vars["name"] = common.CoalesceString(vars["name"], t.name)
-	vars["rack_name"] = common.CoalesceString(vars["rack_name"], t.name)
+
+	// only 3.11.2+ supports rack_name
+	c, err := t.Client()
+	if err == nil {
+		s, _ := c.SystemGet()
+
+		if s.Name != "" {
+			rv, _ := ConvertToReleaseVersion(s.Version)
+
+			if rv != nil {
+				if rv.Minor > MINOR_VERSION_RACK_NAME_SUPPORT ||
+					(rv.Minor == MINOR_VERSION_RACK_NAME_SUPPORT && rv.Revision >= PATCH_VERSION_RACK_NAME_SUPPORT) {
```

Comment:
i can see this check is occurring in several places. i think it would be better to put this check into a helper/util function